evm/src/util.rs

```diff
@@ -210,3 +210,25 @@ pub(crate) fn biguint_to_mem_vec(x: BigUint) -> Vec<U256> {
 pub(crate) fn h2u(h: H256) -> U256 {
     U256::from_big_endian(&h.0)
 }
+
+pub(crate) fn get_h160<F: RichField>(slice: &[F]) -> H160 {
+    H160::from_slice(
+        &slice
+            .iter()
+            .rev()
+            .map(|x| x.to_canonical_u64() as u32)
+            .flat_map(|limb| limb.to_be_bytes())
+            .collect_vec(),
+    )
+}
+
+pub(crate) fn get_h256<F: RichField>(slice: &[F]) -> H256 {
```

Comment:
nit: I'd prefer `extract` over `get` but perhaps that's just me. 

A macro here would be nice!

Reply:
> I'd prefer extract over get

Hmm, it seems `get` is a bit more accurate no? extracting makes me think we take a portion of the provided slice, while here we expect the whole slice to be used to recover the hash. But maybe that's my English being plain wrong :D 

> A macro here would be nice!

For inlining? I like the idea, though I think this may be out of scope of this PR, as several other methods in the `util` module could be turned to macros. WDYT?

Reply:
>Hmm, it seems get is a bit more accurate no?

`get` makes far more sense